conseil-lorre/src/test/scala/tech/cryptonomic/conseil/indexer/forking/ForkDetectorSpec.scala

```diff
@@ -0,0 +1,114 @@
+   *
+   * Similarly we can establish how many discarded values (by a `whenever(predicate)` block) are
+   * acceptable, as a multiplier of the overall successes.
+   * E.g. forAll(maxDiscardedFactor(3.0)) means that for 100 successful runs, no more than 300
+   * test values should've been discarded by using `whenever`.
+   */
+
+  property("The fork-detection algorithm should identify the forking point in arbitrary ranges") {
+    forAll { (range: ForkDetectionRange) =>
+      val sut = detector(forkingPoint = range.fork)
+
+      sut.searchForkLevel(range.low, range.high) shouldBe range.fork
+    }
+  }
+
+  property("The fork detection level check should return consistent results below the forking point") {
+    forAll(maxDiscardedFactor(3.0)) { (range: ForkDetectionRange, level: Long) =>
+      whenever(level < range.fork) {
+        val sut = detector(forkingPoint = range.fork)
+        sut.checkOnLevel(level) shouldBe ForkDetector.SameId
+      }
+    }
+  }
+
+  property("The fork detection level check should return consistent results above the forking point") {
+    forAll(maxDiscardedFactor(3.0)) { (range: ForkDetectionRange, level: Long) =>
+      whenever(level >= range.fork) {
+        val sut = detector(forkingPoint = range.fork)
+        sut.checkOnLevel(level) shouldBe ForkDetector.ForkedId
+      }
+    }
+  }
+
+  /** Defines all necessary data, generators, dummy objects to run the test */
+  private object Fixtures {
+    /* we only use two possible block id values, since we only need to
+     * test if they match or not for a single level, but care not what are
+     * the actual values.
+     * The dummy services that return block-ids per level, will be defined
+     * as to provide the same block id value for inputs below a specific level
+     * and different ids for inputs above such level, where such level is when
+     * the fork is supposed to have happened.
+     * We chose to encode two alternate values as an `Either` type since we don't care about
+     * the actual content, only to differentiate whether it's one or the other.
+     * Thus we can put a Unit value in the Either, which doesn't force any
+     * choice of value as content, being as arbitrary as any other.
+     */
+    type BlockId = Either[Unit, Unit]
+    val blockA: BlockId = Left(())
+    val blockB: BlockId = Right(())
+
+    /* Our test detector will always return the same id for the indexer.
+     * In the case of the node search, while up-to the given [[forkingPoint]]
+     * we'll receive the same indexer id as the node one, and the other
+     * one above the fork point.
+     *
+     * The effect wrapping the results used for the test is [[cats.Id]],
+     * which is nothing but an alias for an empty wrapper, which corresponds
+     * directly to the underlying content type,
+     *  e.g. a cats.Id[Int] = Int
+     */
+    def detector(forkingPoint: Long) = new ForkDetector[cats.Id, BlockId](
+      indexerSearch = (_) => blockA,
```

Comment:
IntelliJ indicated the parentheses are not required.

Reply:
thanks, fixed